File: backend/src/main/kotlin/com/musicstreaming/models/Track.kt
package com.musicstreaming.models

import kotlinx.serialization.Serializable

@Serializable
data class Track(
    val id: String,
    val title: String,
    val artist: String,
    val durationSeconds: Long,
    val provider: ProviderType,
    val thumbnailUrl: String? = null,
    val externalUrl: String? = null
) {
    init {
        require(id.isNotBlank()) { "Track id cannot be blank" }
        require(title.isNotBlank()) { "Track title cannot be blank" }
        require(artist.isNotBlank()) { "Track artist cannot be blank" }
        require(durationSeconds >= 0) { "Track duration must be non-negative, got: $durationSeconds" }
    }
}

File: backend-node/src/models/Track.ts
import { ProviderType } from './ProviderType';

export interface Track {
  id: string;
  title: string;
  artist: string;
  durationSeconds: number;
  provider: ProviderType;
  thumbnailUrl?: string;
  externalUrl?: string;
}

export function assertTrack(track: Track): void {
  if (!track.id || !track.title || !track.artist) {
    throw new Error('Invalid track payload');
  }
  if (track.durationSeconds < 0) {
    throw new Error('Track duration must be non-negative');
  }
}
